src/editors/containers/ProblemEditor/components/EditProblemView/SettingsWidget/hooks.js

```diff
@@ -210,7 +210,9 @@ export const typeRowHooks = ({
     if (typeKey === ProblemTypeKeys.DROPDOWN) {
       if (correctAnswerCount > 1) {
         answers.forEach(answer => {
-          updateAnswer({ ...answer, correct: false });
+          if (answer.correct) {
+            updateAnswer({ ...answer, correct: false });
+          }
```

Comment:
Do you think extracting line 211-217 into a function named something like `clearPreviouslySelectedAnswers()` would be helpful to clarify what the code does? That would also help to reduce nested if statements for readability.

Reply:
I think it makes sense for me to add comments rather than a separate function based on the way that the code is currently structured for all the setting hooks.

Reply:
nit: what about the structure of the code for the settings hooks prevents this? The settings hooks are all about extracting functions, and there is obviously a (very small) problem with the readability of the code here. If you don't like the extracted function in the module scope, a nice pattern in this case is just to declare the new function inside of the upper typeRowHooks function - we do that everywhere in these hooks files.
Code should be largely self-documenting and use explanatory variables, according to the clean code principles. In terms of clean code, comments are used to add extra information that does not restate what the code does, but clarifies intent or warns of consequences (for example that this bug may appear). Nested if-statements are usually not recommended because they increase the cognitive load for understanding what code does a lot (I usually avoid any level of nesting of if-statements.)